cogs/afterhours/afterhours.py

```diff
@@ -0,0 +1,145 @@
+            handler = logging.FileHandler(
+                filename=str(saveFolder) + "/info.log", encoding="utf-8", mode="a"
+            )
+            handler.setFormatter(
+                logging.Formatter("%(asctime)s %(message)s", datefmt="[%d/%m/%Y %H:%M:%S]")
+            )
+            self.logger.addHandler(handler)
+
+    async def getContext(self, channel: discord.TextChannel):
+        """Get the Context object from a text channel.
+
+        Parameters
+        ----------
+        channel: discord.TextChannel
+            The text channel to use in order to create the Context object.
+
+        Returns
+        -------
+        ctx: Context
+            The context needed to send messages and invoke methods from other cogs.
+        """
+        ctxGuild = channel.guild
+        ctxChannelId = await self.config.guild(ctxGuild).channelId()
+        ctxChannel = discord.utils.get(ctxGuild.channels, id=ctxChannelId)
+        if not ctxChannel:
+            self.logger.error("Cannot find channel to construct context!")
+            return None
+        async for message in ctxChannel.history(limit=1):
+            lastMessage = message
+        return await self.bot.get_context(lastMessage)
+
+    async def makeStarboardChanges(
+        self, ctx: Context, channel: discord.abc.GuildChannel, remove=False
+    ):
+        """Apply Starboard changes.
+
+        Parameters
+        -----------
+        ctx: Context
+            The Context object in order to invoke commands
+        channel: discord.abc.GuildChannel
+            The channel to apply Starboard changes to.
+        remove: bool
+            Indicate whether we want to remove the changes. Defaults to False.
+        """
+        self.logger.info("Applying/removing Starboard exceptions, remove=%s", remove)
+        sbCog = self.bot.get_cog("Starboard")
+        if not sbCog:
+            self.logger.error("Starboard not loaded. skipping")
+            return
+
+        try:
+            starboard = sbCog.starboards[ctx.guild.id]["highlights"]
+        except KeyError:
+            self.logger.error("Cannot get the starboard!")
+
+        if remove:
+            await ctx.invoke(sbCog.blacklist_remove, starboard=starboard, channel_or_role=channel)
+        else:
+            await ctx.invoke(sbCog.blacklist_add, starboard=starboard, channel_or_role=channel)
+
+    async def notifyChannel(self, ctx, remove=False):
+        if remove:
+            await ctx.send(f":information_source: **{AH_CHANNEL} removed, removing exceptions**")
+        else:
+            await ctx.send(f":information_source: **{AH_CHANNEL} created, adding exceptions**")
+
+    @commands.Cog.listener("on_guild_channel_create")
+    async def handleChannelCreate(self, channel: discord.abc.GuildChannel):
+        """Listener to see if we need to add exceptions to a channel"""
+        self.logger.info(
+            "Channel creation has been detected. Name: %s, ID: %s", channel.name, channel.id
+        )
```

Comment:
Type check to see if channel is `discord.TextChannel`, and if not, do an early return.